tests/correctness/sync/sync-two-barriers.futil

```diff
@@ -0,0 +1,61 @@
+import "primitives/core.futil";
+import "primitives/sync.futil";
+
+component main() -> () {
+  cells {
+    @external out = std_mem_d1(32, 5, 3);
+    val = std_reg(32);
+    add_0 = std_add(32);
+    addr = std_reg(3);
+    add_1 = std_add(3);
+    lt = std_lt(3);
+    no_use = std_reg(32);
+  }
+
+  wires {
+    group no_op {
+      no_use.in = 32'd0;
+      no_use.write_en = 1'd1;
+      no_op[done] = no_use.done;
+    }
+
+    group incr_val {
+      add_0.left = val.out;
+      add_0.right = 32'd1;
+      val.in = add_0.out;
+      val.write_en = 1'd1;
+      incr_val[done] = val.done;
+    }
+
+    group reg_to_mem {
+      out.write_en = 1'd1;
+      out.write_data = val.out;
+      out.addr0 = addr.out;
+      reg_to_mem[done] = out.done;
+    }
+
+    group incr_idx {
+      add_1.left = addr.out;
+      add_1.right = 3'd1;
+      addr.in = add_1.out;
+      addr.write_en = 1'd1;
+      incr_idx[done] = addr.done;
+    }
+
+  }
+
+  control {
+    par {
+        seq {
```

Comment:
I think again with this, to make sure that the barrier is doing the synchronization we expect, we should make sure that we read and write values between the barriers and observe them in different threads afterwards

Reply:
I'm not sure if this is resolved either. I would expect something like:
```
thread 1: wr(x); sync; no_op; sync; rd(y)
thread 2: sync; rd(x); wr(y); sync
```